app/src/main/java/com/example/reduxtestapp/data/repository/country/CountryRepositoryImpl.kt

```diff
@@ -20,9 +24,17 @@ class CountryRepositoryImpl (
         )
     }
 
+    private suspend fun getCountryListCached(cacheKey: String, countryListSource: suspend () -> List<CountryDto>): List<CountryDto> {
+        val cachedValue = cacheManager.get(cacheKey)
+        return cachedValue ?: countryListSource().also { cacheManager.put(cacheKey, it) }
+    }
```

Comment:
make a generic method like this in Cache so that you can use it everywhere (you'll have something similar wherever you use cache)